indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -422,18 +432,19 @@
       (index-variable context (get-in association [:extra-fields :variable-concept-id]) {}))))
 
 (defmethod index-concept :tag-association
-  [context concept parsed-concept options]
+  [context concept _parsed-concept options]
   (index-associated-collection context concept options))
 
 (defmethod index-concept :variable-association
-  [context concept parsed-concept options]
+  [context concept _parsed-concept options]
   (index-associated-collection context concept options)
   (index-associated-variable context concept {})
   (index-associated-variable context concept {:all-revisions-index? true}))
 
 (defmethod index-concept :service-association
   [context concept parsed-concept options]
-  (index-associated-collection context concept options))
+  (index-associated-collection context concept options)
+  (index-concept context concept parsed-concept {:all-revisions-index? true}))
```

Comment:
I'm not sure whether we should be updating associations for the all revisions index. @ygliuvt - do you know what the use case is for associations in the all revisions index?

Also shouldn't this code be `index-associated-service` to the regular index and the all revisions-index? We don't actually index associations, we index the thing they are associated to. Variables does what I expect in lines 440-442 if we do update the all-revisions index for associations (and that would also answer my question about whether we should be indexing to the all revisions index).

This seems to also indicate there's a missing test for creating a service association and verifying that association can be found in the services search route. Do we track associations for services to collections, or are we only tracking associations from collections to services?

Reply:
I had misunderstood what has happening here; I've just walked through the code again and made corresponding fixes/change. We should continue this conversation based upon those recent changes ...